other/field-types/joinByArray.md

```diff
@@ -0,0 +1,115 @@
+
+{% hint style='info' %}
+**For performance, it is strongly recommended that you set a projection filter** via the `filters` option, limiting the amount of information fetched about each related doc. You may also call other [cursor filters](../../modules/apostrophe-docs/server-apostrophe-cursor.md) by setting subproperties of the `filters` property. This is a useful way to limit the acceptable choices for the join.
+{% endhint %}
+
+## Example 1
+
+```javascript
+{
+  name: '_fabrics',
+  label: 'Fabrics',
+  // This is optional since the name of our join matches the name of the
+  // other type, if the names don't match it is mandatory
+  withType: 'fabric',
+  type: 'joinByArray',
+  filters: {
+    // Fetch just enough information
+    projection: {
+      title: 1,
+      _url: 1
+    }
+  }
+}
+```
+
+{% hint style='info' %}
+You could separately specify `slug`, `type`, and `tags`, but using `_url` specifies sufficient information to generate the `_url` property with less writing. You'll see this in the next example as well.
+{% endhint %}
+
+
+## Relationship properties and `joinByArray`
+
+Sometimes, the relationship between the two objects has properties of its own. For example, the relationship between a person and a department might have a `jobTitle` property. Yes, a person can have more than one job title!
+
+You can express these relationships by using the `relationship` property, which allows you to specify a schema for the relationship. _You may use most schema field types, however joins are not permitted inside the schema for the relationship. If the relationship is overly complex, it is recommended that you consider treating it as a third type of object related to both of the other two._
+
+When you specify the `relationship` property, you **may** also specify `relationshipsField`, a property name to store the relationships in. If you do not specify this property, it is set automatically. For instance, if the join is named `_departments`, the relationships will be stored in `departmentsRelationships`.
+
+### Example 2
+
+```javascript
+{
+  name: '_departments',
+  label: 'Departments',
+  // We could skip this since it is the same as the name of the join,
+  // plus an s. Since we specified it, it must match the `name` property
+  // of another type (it'll be singular)
+  withType: 'department',
+  type: 'joinByArray',
+  filters: {
+    // Fetch just enough information
+    projection: {
+      title: 1,
+      _url: 1
+    }
+  },
+  relationship: [
+    {
+      name: 'jobTitle',
+      label: 'Job Title',
+      type: 'string'
+    }
+  ]
+}
+```
+
+Since there is a relationship, the `_departments` property will be an array of objects with `item` and `relationship` properties. The `item` property will be the actual department, and the `relationship` property will contain the relationship fields, which are unique to this pairing.
```

Comment:
Since there is a relationship, when the join is actually fetched and attached to an object, the...

Reply:
Updated.